capsule/src/main/java/Capsule.java

```diff
@@ -3693,20 +3702,39 @@ private static boolean shouldExtractFile(String fileName) {
         return true;
     }
 
-    private List<Path> listJar(Path jar, String glob, boolean regular) {
-        final long start = clock();
-        final List<Path> res = new ArrayList<>();
-        final Pattern p = Pattern.compile(globToRegex(glob));
-        try (ZipInputStream zis = openJarInputStream(jar)) {
-            for (ZipEntry entry; (entry = zis.getNextEntry()) != null;) {
-                if ((!regular || !entry.isDirectory()) && p.matcher(entry.getName()).matches())
-                    res.add(path(entry.getName())); // new URL("jar", "", jar + "!/" + entry.getName())
-            }
-        } catch (IOException e) {
+    private List<Path> listJar(Path jar, String glob, boolean regular)
+    {
+        long start = clock();
+        try (FileSystem fileSystem = FileSystems.newFileSystem(jar, null))
+        {
+            Matcher matcher = Pattern.compile(globToRegex(glob)).matcher("");
+            return StreamSupport.stream(fileSystem.getRootDirectories().spliterator(), false)
+                .flatMap(rootDir -> {
+                    try
+                    {
+                        return Files.walk(rootDir);
+                    }
+                    catch (IOException e)
+                    {
+                        throw new RuntimeException(e);
+                    }
+                })
+                .filter(p -> !regular || Files.isRegularFile(p))
+                .map(Path::toString)
+                .map(p -> p.startsWith("/") ? p.substring(1) : p)
```

Comment:
Could this be a backslash on windows?
PS: what is the purpose of this check and strip?

Reply:
We are browsing a Jar content, so all filenames are separated with '/' whatever platform we are working with.

Beside filenames in jars can start with a '/' or not; however the path() function turning Strings into (relative) Paths uses the default filesystem provider to build the path, so starting with a '/' can have a special meaning (absolute path starting from unix root) which we definitely don't want here.

So we normalize all  jar entries paths without the '/', to end up with the same result has the previous implementation